### ISSUE TITLE
potential for uv or pixi/conda support?

### ISSUE THREAD

**Issue by gregorywaynepower** [kraina-ai/quackosm#142]
I've really enjoyed working with your package and it really is fantastic.

Do you think you'll be able to support [uv](https://docs.astral.sh/uv/) or adding this to [conda-forge's feedstock](https://github.com/conda-forge/staged-recipes?tab=readme-ov-file#about) so conda or pixi users could install your package in a more isolated manner?

- [x] Support for `uv`: Just use `uv tool install QuackOSM[cli] --prerelease=allow`
- [ ] Support for conda-forge through [conda-forge's feedstock](https://github.com/conda-forge/staged-recipes?tab=readme-ov-file#about)

**Comment by RaczeQ** [comment]
Hi @gregorywaynepower, thank you for this suggestion!
I have to admit that I have never tried publishing any library outside of PyPI, so I'm not familiar with the process, but I might take a shot and try publishing them there.

**Comment by gregorywaynepower** [comment]
Good to see that `quackosm` itself installs on Windows via `uv tool install quackosm`! I haven't the same amount of luck with `uv tool install quackosm[cli]`

[uv_error_log.txt](https://github.com/user-attachments/files/16752644/uv_error_log.txt)

**Comment by RaczeQ** [comment]
It seems like an error with the DuckDB wheels. I can also see that it's probably trying to install an older version of QuackOSM, because now minimal DuckDB version is `0.10.2` (since [0.6.1](https://github.com/kraina-ai/quackosm/releases/tag/0.6.1)). Maybe trying to install a newer version will work?

I tried to run uv with the `cli` extras on Ubuntu WSL machine (have to use `--prerelease=allow`):
```console
/mnt/c/Development/Python/uv-test$ uv add QuackOSM[cli] --prerelease=allow
Ignoring existing lockfile due to change in pre-release mode: `if-necessary-or-explicit` vs. `allow`
Resolved 52 packages in 1.49s
warning: The package `typer==0.12.5` does not have an extra named `all`
   Built uv-test @ file:///mnt/c/Development/Python/uv-test
   Built python-geohash==0.8.5
Prepared 16 packages in 7.88s
Uninstalled 6 packages in 3.92s
░░░░░░░░░░░░░░░░░░░░ [0/16] Installing wheels...                                                                                                                                            warning: Failed to hardlink files; falling back to full copy. This may lead to degraded performance.
         If the cache and target directories are on different filesystems, hardlinking may not be supported.
         If this is intentional, set `export UV_LINK_MODE=copy` or use `--link-mode=copy` to suppress this warning.
Installed 16 packages in 6.19s
 - beautifulsoup4==4.12.3
 + beautifulsoup4==4.13.0b2
 + click==8.1.7
 - duckdb==1.0.0
 + duckdb==1.0.1.dev4693
 + future==1.0.0
 - geoarrow-rust-core==0.2.0
 + geoarrow-rust-core==0.3.0a1
 + geographiclib==2.0
 + geopy==2.4.1
 + h3==4.0.0b5
 - numpy==2.0.1
 + numpy==2.0.2
 + python-geohash==0.8.5
 - rich==13.7.1
 + rich==13.8.0
 + s2==0.1.9
 + s2sphere==0.2.5
 + shellingham==1.5.4
 + typer==0.12.5
 ~ uv-test==0.1.0 (from file:///mnt/c/Development/Python/uv-test)
```

**Comment by gregorywaynepower** [comment]
Fantastic! I'm able to install this on natively Windows with `uv tool install QuackOSM[cli] --prerelease=allow` on Windows too!

Want me to keep this open for the conda-forge portion?

**Comment by RaczeQ** [comment]
Yeah, let's leave it open for the conda-forge part. I'm happy that `uv` works on your end. And thanks again for raising this issue, since I'm not the conda user myself 😉

**Comment by RaczeQ** [comment]
The recipe is currently in review: https://github.com/conda-forge/staged-recipes/pull/30337

**Comment by gregorywaynepower** [comment]
Fantastic news @RaczeQ !

**Comment by RaczeQ** [comment]
Library available on Conda-Forge: https://anaconda.org/conda-forge/quackosm
